### PR TITLE
Add external API server

### PR DESCRIPTION
**Pull request by cdonati**
This introduces a separate process for handling API requests from runtime processes. For now, it just handles App Identity calls for Python runtimes.

### CODE REVIEW

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/4377/

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/4380/

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/4486/

File: AppManager/app_manager_server.py [hunk 1]
```diff
@@ -158,6 +172,85 @@ def add_routing(version_key, port):
...
+def ensure_api_server(project_id):
+  """ Make sure there is a running API server for a project.
+
+  Args:
+    project_id: A string specifying the project ID.
+  Returns:
+    An integer specifying the API server port.
+  """
+  global api_servers
+  logging.warning('api_servers: {}'.format(api_servers))
```
Comment (whoarethebritons) [inline AppManager/app_manager_server.py]: should this be demoted from warning to info or debug

File: AppManager/app_manager_server.py [hunk 2]
```diff
@@ -158,6 +172,85 @@ def add_routing(version_key, port):
...
+
+  global api_servers
+  monit_entries = yield monit_operator.get_entries()
+  server_entries = [api_server_info(entry) for entry in monit_entries
+                    if entry.startswith(API_SERVER_PREFIX)]
+
+  for project_id, port in server_entries:
+    api_servers[project_id] = port
+
+  logging.warning('api_servers: {}'.format(api_servers))
```
Comment (whoarethebritons) [inline AppManager/app_manager_server.py]: same here

Comment (cdonati) [reply AppManager/app_manager_server.py]: Yes, I meant to remove those.

**Comment by cdonati** [comment]
@jeanleonov Yes, I had forgotten to update the package at runtime.

File: AppManager/app_manager_server.py [hunk 3]
```diff
@@ -54,6 +55,13 @@
 from appscale.common.monit_interface import ProcessNotFound
 
 
+# The location of the API server start script.
+API_SERVER_LOCATION = os.path.join('/', 'opt', 'appscale_api_server', 'bin',
```
[the remark this answers is not in the document]
Comment (cdonati) [reply AppManager/app_manager_server.py]: The API server has its own virtualenv, so the script is not on the path.

**Comment by cdonati** [comment]
Autoscaled deployments will require a new image after upgrading since this requires a new system package and virtualenv.

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/4579/

**Comment by cdonati** [comment]
App Identity Demo:
https://github.com/AppScale/hawkeye/pull/125 with master: https://ocd.appscale.com:8080/job/Hawkeye/13137/
https://github.com/AppScale/hawkeye/pull/125 with branch: https://ocd.appscale.com:8080/job/Hawkeye/13136/

**Pull request merged**